src/codegen.c

```diff
@@ -367,6 +368,14 @@ void code_generate()
             if (dump_ir == 1)
                 printf("    x%d /= x%d", dest_reg, OP_reg);
             break;
+        case OP_mod:
+            emit(__div(__AL, OP_reg + 1, OP_reg, dest_reg));
```

Comment:
Both `div` and `mul` instruction are heavier. Instead, you can add shortcut for power-of-2. That is, convert the original modulo operation into `n & (d - 1)` if `d` is power-of-2. You should implement the shortcut in `src/cfront.c`.

Reply:
I think only global expression can do this shortcut because `shecc` can't determine the value of operand in local expression before execution time. Can we ignore optimization for local expression in this patch?

BTW, relative [issue#28](https://github.com/jserv/shecc/issues/28) and detailed information was sent.

Reply:
Sure, let's merge this change first. Do append the comments with `TODO: optimize ...`